.github/workflows/build-and-run.yml

```
@@ -0,0 +1,109 @@
+name: Build-And-Run Tests
+
+on: [push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_and_test:
+    env:
+      ALL_EXCLUDE: stage_dem|ionosphere|background|pybind.unwrap.phass
+
+    defaults:
+      run:
+        shell: bash -l {0}
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - label: Linux
+            runner: ubuntu-latest
+            ctest_exclude: -E ".*($ALL_EXCLUDE)"
+          - runner: macos-latest
+            label: macOS
+            ctest_exclude: -E ".*($ALL_EXCLUDE|raster|unwrap.snaphu)"
+        deps:
+          - label: Latest
+            env-file: env/environment.yml
+            extra-specs: ""
+        build-type: [ Release, Debug ]
+        include:
+          - os:
+              label: Linux
+              runner: ubuntu-latest
+              ctest_exclude: -E ".*($ALL_EXCLUDE)"
+            deps:
+              label: Minimum
+              env-file: env/minimum-reqs.yml
+              extra-specs: "gsl"
+            build-type: Release
+
+    name: Build And Test - ${{ matrix.os.label }} ${{ matrix.deps.label }} ${{ matrix.build-type }}
+    runs-on: ${{ matrix.os.runner }}
+    steps:
+      # Typical github repo checkout step
+      - name: Github Repo Checkout
+        uses: actions/checkout@v2
+
+      # Prints the variables for this run to simplify debugging
+      - name: Print Run Variables
+        run: |
+          echo Runner OS: ${{ matrix.os.runner }}
+          echo Environment: ${{ matrix.deps.env-file }}
+          echo Build Type: ${{ matrix.build-type }}
+          echo ctest exclude command: ${{ matrix.os.ctest_exclude }}
+          echo Depenencies: ${{ matrix.deps.extra-specs }}
+
+      # Set the conda environment up using Mamba and the dependencies for this run
+      - name: Setup Environment
+        uses: mamba-org/provision-with-micromamba@main
+        with:
+          environment-file: ${{ matrix.deps.env-file }}
+          environment-name: isce3-ci-sandbox
+          extra-specs: ${{ matrix.deps.extra-specs }}
+
+      # Preset environment variables in the conda environment for later steps
+      - name: Environment Variables Preset
+        run: |
+          export ISCE3_PREFIX=${GITHUB_WORKSPACE}/install
+          conda env config vars set ISCE3_PREFIX=$ISCE3_PREFIX
+          conda env config vars set PYTHONPATH=$PYTHONPATH:$ISCE3_PREFIX/packages
+          if test -d "$ISCE3_PREFIX/lib64"; then
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ISCE3_PREFIX/lib64:$CONDA_PREFIX/bin/ld
+          else
+            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ISCE3_PREFIX/lib:$CONDA_PREFIX/bin/ld
```

I don't think it should be necessary to add the `ld` linker itself to `LD_LIBRARY_PATH`

```suggestion
            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ISCE3_PREFIX/lib64
          else
            export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ISCE3_PREFIX/lib
```